File: binary_tree/104_maximum_depth_of_binary_tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func maxDepth(root *TreeNode) int {

    return dfs(root, 0)

}

func dfs(node *TreeNode, depth int) int {

    if node == nil {
        return depth
    }

    leftDepth := dfs(node.Left, depth+1)
    rightDepth := dfs(node.Right, depth+1)

    if leftDepth > rightDepth {
        return leftDepth
    } else {
        return rightDepth
    }

}

File: binary_tree/104_maximum_depth_of_binary_tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function(root) {

    function dfs(node, depth){

        if(!node) return depth

        let leftDepth = dfs(node.left, depth+1)
        let rightDepth = dfs(node.right, depth+1)

        return Math.max(leftDepth, rightDepth)
    }

    return dfs(root, 0)
};